test/functional/feature_reindex_readonly.py

```
@@ -0,0 +1,90 @@
+        self.generateblock(self.nodes[0], output=f"raw({opreturn}{nulldata})", transactions=[])
+        block_count = self.nodes[0].getblockcount()
+        self.stop_node(0)
+
+        assert (self.nodes[0].chain_path / "blocks" / "blk00000.dat").exists()
+        assert (self.nodes[0].chain_path / "blocks" / "blk00001.dat").exists()
+
+        self.log.debug("Make the first block file read-only")
+        filename = self.nodes[0].chain_path / "blocks" / "blk00000.dat"
+        filename.chmod(stat.S_IREAD)
+
+        undo_immutable = lambda: None
```

Ruff (0.12.2)

Do not assign a `lambda` expression, use a `def`

Rewrite `undo_immutable` as a `def`

(E731)